### PR TITLE
feat: add sources validation api (MAPCO-4505)

### PR DESCRIPTION
**Pull request by asafMasa**
| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✖                                                                        |
| New feature     | ✔                                                                        |
| Breaking change | ✖                                                                        |
| Deprecations    | ✖                                                                        |
| Documentation   | ✖                                                                        |
| Tests added     | ✔                                                                        |
| Chore            | ✖                                                                       |

Add sources validation + tests

### CODE REVIEW

File: src/model/models/modelManager.ts [hunk 1]
```diff
@@ -29,6 +29,46 @@ export class ModelManager {
...
+    });
+
+    const resultModelPathValidation = this.validator.validateModelPath(payload.modelPath);
+    if (typeof resultModelPathValidation === 'string') {
+      return {
+        isValid: false,
+        message: resultModelPathValidation,
+      };
+    }
+    payload.modelPath = this.getAdjustedModelPath(payload.modelPath);
```
Comment (asafMasa) [inline src/model/models/modelManager.ts:48]: change order

File: src/model/models/modelManager.ts [hunk 2]
```diff
@@ -29,6 +29,46 @@ export class ModelManager {
...
+    const sourcesValidationResponse = await this.validator.sourcesValid(payload);
```
Comment (asafMasa) [inline src/model/models/modelManager.ts:49]: send with updated model path

File: src/validator/validationManager.ts [hunk 3]
```diff
@@ -41,6 +48,51 @@ export class ValidationManager {
...
+    const isValidTilesetContent = await this.isValidTilesetContent(tilesetLocation, isValidTilesetContentResponse);
+    if (!isValidTilesetContent) {
+      return {
+        isValid: false,
+        message: isValidTilesetContentResponse.message,
+      };
+    }
+
+    // TODO: refactor in next PR in order to return bool and better message
+    const result = this.validateFootprint(isValidTilesetContentResponse.polygon as Polygon);
```
Comment (asafMasa) [inline src/validator/validationManager.ts:83]: move to tileset content

Comment (asafMasa) [reply src/validator/validationManager.ts]: fixed

File: .vscode/launch.json [hunk 4]
```diff
@@ -29,6 +29,32 @@
         "TELEMETRY_METRICS_URL": "http://localhost:55681/v1/metrics"
         // "OPENAPI_FILE_PATH": "./openapi3.yaml"
       }
+    },
```
Comment (netanelC) [inline .vscode/launch.json:32]: Unnecessary change

File: src/model/routes/modelRouter.ts [hunk 5]
```diff
@@ -7,6 +7,8 @@ const modelRouterFactory: FactoryFunction<Router> = (dependencyContainer) => {
   const controller = dependencyContainer.resolve(ModelController);
 
   router.post('/', controller.createModel);
+  router.post('/validate', controller.validate.bind(controller));
+  router.post('/validateSources', controller.validateSources.bind(controller));
```
Comment (netanelC) [inline src/model/routes/modelRouter.ts:11]: Consider change the route
```suggestion
  router.post('/validate/sources', controller.validateSources.bind(controller));
```

File: src/validator/validationManager.ts [hunk 6]
```diff
@@ -298,7 +341,7 @@ export class ValidationManager {
       const coverage = (100 * areaFootprint) / areaCombined;
 
       if (coverage < this.limit) {
-        return `The footprint is not intersected enough with the model, the coverage is: ${coverage}% when the minimum coverage is ${this.limit}%`;
+        return `The footprint intersectection with the model doesn't reach minimum required threshhold, the coverage is: ${coverage}% when the minimum coverage is ${this.limit}%`;
```
Comment (netanelC) [inline src/validator/validationManager.ts:344]: Typo
```suggestion
        return `The footprint intersection with the model doesn't reach minimum required threshold, the coverage is: ${coverage}% when the minimum coverage is ${this.limit}%`;
```

File: src/validator/validationManager.ts [hunk 7]
```diff
@@ -374,4 +417,83 @@ export class ValidationManager {
...
+        error,
+      });
+      isValidTilesetContentResponse.message = message;
+      return false;
+    }
+    return true;
+  }
+
+  @withSpanV4
+  private async validateFilesExist(fullPath: string): Promise<boolean> {
```
Comment (netanelC) [inline src/validator/validationManager.ts:469]: You wrote a new function that simply execute one fs operation. Why not just call the operation without set it in a new function?

File: tests/helpers/helpers.ts [hunk 8]
```diff
@@ -72,7 +73,8 @@ export const createModelPath = (modelName = 'Sphere'): string => {
 };
 
 export const createMountedModelPath = (modelName = 'Sphere'): string => {
-  return `${pvPath}/${modelName}`;
+  const mountedPath = join(`${pvPath}`, `${modelName}`);
```
Comment (netanelC) [inline tests/helpers/helpers.ts:76]: Unnecessary quotes
```suggestion
  const mountedPath = join(pvPath, modelName);
```

File: tests/unit/validator/validationManager.spec.ts [hunk 9]
```diff
@@ -438,7 +521,6 @@ describe('ValidationManager', () => {
       };
       catalogMock.isProductIdExist.mockResolvedValue([payload.metadata.productId]);
       lookupTablesMock.getClassifications.mockResolvedValue([payload.metadata.classification]);
```
Comment (netanelC) [inline tests/unit/validator/validationManager.spec.ts:523]: Undo the empty line removal

**Review by netanelC — CHANGES_REQUESTED** [review]

File: .vscode/launch.json (hunk 4 above)
Comment (asafMasa) [reply .vscode/launch.json:32]: removed

File: src/model/routes/modelRouter.ts (hunk 5 above)
Comment (asafMasa) [reply src/model/routes/modelRouter.ts:11]: why? i think it's better not to put both validation in same hirerachy

File: src/validator/validationManager.ts (hunk 7 above)
Comment (asafMasa) [reply src/validator/validationManager.ts:469]: 2 reasons:
* because i wan't the logs (before and after) with same paramaters
* The function contains 10 rows that will make the caller code "ugly";

File: tests/unit/validator/validationManager.spec.ts (hunk 9 above)
Comment (asafMasa) [reply tests/unit/validator/validationManager.spec.ts:523]: done

File: src/model/routes/modelRouter.ts (hunk 5 above)
Comment (netanelC) [reply src/model/routes/modelRouter.ts:11]: Why not? nevertheless, it is not the best practice to use camel case in URIs.
See: https://restfulapi.net/resource-naming/

File: src/validator/validationManager.ts (hunk 7 above)
Comment (netanelC) [reply src/validator/validationManager.ts:469]: OK
